kube-runtime/src/watcher.rs

```diff
@@ -48,28 +48,24 @@ pub enum Event<K> {
     Delete(K),
     /// The watch stream was restarted.
     ///
-    /// If using the `ListWatch` strategy, `RestartPage` events will follow this event.
-    /// If using the `StreamingList` strategy, this event will be followed by `RestartApply` or `RestartDelete` events.
-    RestartInit,
-    /// A page of objects was received during the restart.
+    /// If using the `ListWatch` strategy, this is a relist, and indicates that `InitlPage` events will follow.
+    /// If using the `StreamingList` strategy, this event will be followed by `InitApply` + `InitDelete` events.
+    Init,
+    /// A page of objects was received during `Init` with the `ListWatch` strategy.
     ///
-    /// This event is only returned when using the `ListWatch` strategy.
+    /// Any objects that were previously [`Applied`](Event::Applied) but are not listed in any of
+    /// the `InitPage` events should be assumed to have been [`Deleted`](Event::Deleted).
+    InitPage(Vec<K>),
+    /// An object received during `Init` with the `StreamingList` strategy.
     ///
-    /// Any objects that were previously [`Applied`](Event::Applied) but are not listed in any of the pages
-    /// should be assumed to have been [`Deleted`](Event::Deleted).
-    RestartPage(Vec<K>),
-    /// An object was added or modified during the initial watch.
-    ///
-    /// This event is only returned when using the `StreamingList` strategy.
-    RestartApply(K),
-    /// An object was deleted during the initial watch.
-    ///
-    /// This event is only returned when using the `StreamingList` strategy.
-    RestartDelete(K),
-    /// The restart is complete.
+    /// Any objects that were previously [`Applied`](Event::Applied) but are not listed in any of
+    /// the `InitAdd` events should be assumed to have been [`Deleted`](Event::Deleted).
+    InitApply(K),
+    /// The initialisation is complete.
     ///
     /// Should be used as a signal to replace the store contents atomically.
-    Restart,
+    /// No more `InitPage` / `InitApply` events will happen until the next `Init` event.
+    Ready,
```

Comment:
```suggestion
    InitDone,
```

For consistency with the other init variants.

Reply:
Could do, but thinking a bit more, wdyt about this counter-proposal:

```rust
  Reset,
  Initial(K),
  Page(Vec<K>),
  Ready
```

Reply:
I find that to be even more confusing:

1. `Initial` and `Page` don't _feel_ related at all
2. `Reset` feels like "perform the whole reset", not "initiate the reset protocol"
3. `Ready`.. from/for what?

Reply:
Fair points. Perhaps I put too much value on brevity.

Have applied suggestion in https://github.com/kube-rs/kube/pull/1499/commits/3d48ad2f28ca0f7a50acb0ff2683377e27aa9641